### PR TITLE
Simple approach to automated test erc20_btc

### LINKED ISSUE #109: ERC20-BTC example cannot be automatically tested
## Problem

The ERC20-BTC example cannot be _easily_ tested as part of the CI as it requires shell interaction.

## Goal

Automate run of all examples to ensure our user can always run them

## Recommendation

`yes |` command during tests, @D4nte.

## Priority Justification

As it is not tested, we could end up with a non-working example in the repository which could impact COMIT App devs or comit workshops.

### PR DESCRIPTION
**Pull request by bonomat**
Resolves #109 

Piping `y` or similar into a process with the current approach is really hard as to impossible as the process only starts reading sometimes in the middle. 

Hence, I opted for the most obvious solution to enable automated testing for this example.

* https://askubuntu.com/questions/1118109/how-do-i-tell-if-a-command-is-running-or-waiting-for-user-input
* https://stackoverflow.com/questions/52708607/how-to-get-know-if-a-process-is-waiting-on-stdin

### CODE REVIEW

File: create/new_project/demos/erc20_btc/src/index.ts [hunk 1]
```diff
@@ -173,3 +173,9 @@ async function printBalances(actor: Actor) {
 function toNominal(tokenWei: string, decimals: number) {
     return new BigNumber(tokenWei).div(new BigNumber(10).pow(decimals));
 }
+
+function wait_for_confirmation(text: string) {
+    if (process.env.NON_INTERACTIVE !== "true") {
```
Comment (luckysori) [inline create/new_project/demos/erc20_btc/src/index.ts]: :wrench: For clarity; double negations are gnarly.

```suggestion
    if (process.env.INTERACTIVE == "true") {
```

Comment (bonomat) [reply create/new_project/demos/erc20_btc/src/index.ts]: Yea, I was playing with this a bit... the problem is that this var can be undefined and then I still want to be it interactively. Let me push a different version though.

File: create/tests/force_clean_env.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
```
Comment (bonomat) [inline create/tests/force_clean_env.sh]: Once we have this https://github.com/comit-network/create-comit-app/issues/388 we can remove this file again.

Comment (D4nte) [reply create/tests/force_clean_env.sh]: Can you please explain more why this fail is necessary?

File: create/new_project/demos/erc20_btc/src/index.ts [hunk 3]
```diff
@@ -173,3 +173,10 @@ async function printBalances(actor: Actor) {
 function toNominal(tokenWei: string, decimals: number) {
     return new BigNumber(tokenWei).div(new BigNumber(10).pow(decimals));
 }
+
+function wait_for_confirmation(text: string) {
+    if (process.env.NON_INTERACTIVE && process.env.NON_INTERACTIVE === "true") {
```
Comment (D4nte) [inline create/new_project/demos/erc20_btc/src/index.ts]:
```suggestion
    if (process.env.NON_INTERACTIVE) {
```
If you just use `1` or `0` as a value then that should be enough.

File: Makefile [hunk 4]
```diff
@@ -116,6 +116,9 @@ e2e_scenarios:
 	./scripts/tests/start_env.sh
 	./scripts/tests/force_clean_env.sh
 	./create/tests/new.sh
-	./create/tests/demos/btc_eth.sh
+	./create/tests/force_clean_env.sh
```
Comment (D4nte) [inline Makefile]: There is a `wait` and `kill` at the end of each test that prevents the need of such added file.

File: create/new_project/demos/erc20_btc/src/index.ts (hunk 3 above)
Comment (bonomat) [reply create/new_project/demos/erc20_btc/src/index.ts]: unfortunately not. The value will be interpreted as a string.

File: Makefile (hunk 4 above)
Comment (bonomat) [reply Makefile]: Which does not seem to work 100%. 
I think we should resolve #388 and then use this here.

File: create/tests/force_clean_env.sh (hunk 2 above)
Comment (bonomat) [reply create/tests/force_clean_env.sh]: Our environment does not terminate deterministically which will lead to unexpected errors during e2e test execution. If we only run 1 test, then this is obviously not an issue, but with 2 the second might randomly fail. 

See here: 
* https://github.com/comit-network/create-comit-app/runs/400255524
* https://github.com/comit-network/create-comit-app/runs/400238837

File: create/tests/run_demo.sh [hunk 5]
```diff
@@ -92,5 +98,5 @@ kill -s SIGINT $STARTENV_PID;
 wait $STARTENV_PID || true;
```
Comment (D4nte) [inline create/tests/run_demo.sh]: Here, add:
```
COMIT_SCRIPT_PID=$(pgrep comit-scripts)
wait "$COMIT_SCRIPT_PID";
```

That should help avoiding to do the `force-clean-env`

Comment (D4nte) [reply create/tests/run_demo.sh]: Scrap that.. Can't wait for a PID that is not a child.

**Comment by bonomat** [comment]
> Please replace force_clean_env.sh between test with a sleep 2; after wait $STARTENV_PID || true;.

Thanks for your proposal @D4nte . 
I'm afraid that it does not work as proposed :D 
It seems like the issue is actually a different one: 
`$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...

One alternative I could think of is not use yarn but go to the `create-comit-app` binary directly.

**Comment by D4nte** [comment]
> `$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...

This is the core of my explanation... I am surprised it has no effect. Feel free to dismiss my change request and roll with that, I can try to optimise it later.

**Comment by D4nte** [comment]
> > `$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...
> 
> This is the core of my explanation... I am surprised it has no effect. Feel free to dismiss my change request and roll with that, I can try to optimise it later.

Note: it HAS to have an effect because the previous test flow was:

- start-env
- ...
- run example

So surely the first start-env did get killed and this code hasn't changed.

**Comment by bonomat** [comment]
> > > `$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...
> > 
> > 
> > This is the core of my explanation... I am surprised it has no effect. Feel free to dismiss my change request and roll with that, I can try to optimise it later.
> 
> Note: it HAS to have an effect because the previous test flow was:
> 
> * start-env
> * ...
> * run example
> 
> So surely the first start-env did get killed and this code hasn't changed.

mhmm... good point, let me try to investigate

**Comment by bonomat** [comment]
> > > > `$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...
> > > 
> > > 
> > > This is the core of my explanation... I am surprised it has no effect. Feel free to dismiss my change request and roll with that, I can try to optimise it later.
> > 
> > 
> > Note: it HAS to have an effect because the previous test flow was:
> > 
> > * start-env
> > * ...
> > * run example
> > 
> > So surely the first start-env did get killed and this code hasn't changed.
> 
> mhmm... good point, let me try to investigate

the difference is that in these examples we go straight for the binaries, that's why killing the process works:

https://github.com/comit-network/create-comit-app/blob/c766cef070e6ca729942966857ffc15c570d3a5a/scripts/tests/start_env.sh#L5-L13

Last idea is to call `force-clean-env` directly in the test file... which is what I'm doing in the extra shell script.

**Comment by D4nte** [comment]
> > > > > `$STARTENV_PID` is not the process id of `create-comit-app start-env` but of `yarn run start-env`, that's why killing the process has no effect...
> > > > 
> > > > 
> > > > This is the core of my explanation... I am surprised it has no effect. Feel free to dismiss my change request and roll with that, I can try to optimise it later.
> > > 
> > > 
> > > Note: it HAS to have an effect because the previous test flow was:
> > > 
> > > * start-env
> > > * ...
> > > * run example
> > > 
> > > So surely the first start-env did get killed and this code hasn't changed.
> > 
> > 
> > mhmm... good point, let me try to investigate
> 
> the difference is that in these examples we go straight for the binaries, that's why killing the process works:
> 
> https://github.com/comit-network/create-comit-app/blob/c766cef070e6ca729942966857ffc15c570d3a5a/scripts/tests/start_env.sh#L5-L13
> 
> Last idea is to call `force-clean-env` directly in the test file... which is what I'm doing in the extra shell script.

DANG!

**Comment by bonomat** [comment]
@D4nte : check-out the latest version. Maybe that is better :)

**Pull request merged**